api/database/factories/PoolFactory.php

```diff
@@ -50,6 +50,7 @@ public function definition()
         }
 
         $teamId = Team::inRandomOrder()
+            ->whereNull('teamable_id')
             ->limit(1)
             ->pluck('id')
```

Comment:
Constrain it otherwise you can have a `pool->legacyTeam` relation that points at a different pool teamable, which is... wrong

![image](https://github.com/user-attachments/assets/08af6768-aeb2-4e1d-bd1a-3a499fb9a67a)